core/src/main/java/org/apache/druid/timeline/Overshadowable.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.timeline;
+
+/**
+ * Interface to represent a class which can have overshadow relation between its instances.
+ * In {@link VersionedIntervalTimeline}, Overshadowable is used to represent each {@link DataSegment}
+ * which has the same major version in the same time chunk.
+ *
+ * An Overshadowable overshadows another if its root partition range contains that of another
+ * and has a higher minorVersion. For more details, check https://github.com/apache/incubator-druid/issues/7491.
+ */
+public interface Overshadowable<T extends Overshadowable>
+{
+  /**
+   * Returns true this overshadows the given other.
```

Comment:
nit suggest: "Returns true if this segment overshadows the other segment"

Reply:
Done.